### ISSUE TITLE
[Ephemeral buckets] Expiring bucket keys for the RedisBucket

### ISSUE THREAD

**Issue by MicheleMorelli** [vutran1710/PyrateLimiter#187]
Hello @vutran1710,

Thank you very much for this creating and maintaining this module - it's really great 👍 .

This is not really an issue - more of a couple of questions to check I understand the intended usage of the Redis back-end correctly.

- I noticed that the `RedisBucket` implementation does not allow to set the `ttl` for the key (as far as I could tell), so the expiry  for buckets is set to `-1` by default. I was wondering whether this is this this by design, in order to avoid conflicts in case that the period of the Rate is longer than the `ttl` of the bucket's key?
  -  The context of this question is a potential use case where many ephemeral buckets are created dynamically on the fly.

- Am I correct that, in case one wanted to create multiple buckets dynamically using the Redis BE in a distributed context (ie where I could not save all the created buckets in a `dict` in memory), I would have to rely in the `RedisBucket.init()` constructor helper method every time?

To give an example, the implementation of the `BucketFactory.get` method would be something like:
```
    async def get(self, item: RateItem) -> AbstractBucket:
        return await RedisBucket.init(
            rates=rates, redis=redis_connection, bucket_key=some_key
        )
```

In other words, my question is: is using `RedisBucket.init()` as an `UPSERT` function the intended functionality, or are there better ways to fetch an existing bucket from Redis (other than doing so directly using the Redis SDK, of course)?

Again, thanks for the great library!

**Comment by vutran1710** [comment]
@MicheleMorelli for dynamic bucket creation, you can refer to the example here https://github.com/vutran1710/PyrateLimiter?tab=readme-ov-file#creating-buckets-dynamically

**Comment by MicheleMorelli** [comment]
Thanks @vutran1710! 
I saw that example, but I am not sure that it would work reliably in a distributed environment with multiple instance of the app, as it relies on the `self.buckets = {}` being stored in memory (so each instance of the app would have its own version of `buckets`). 

I can think of a couple of ways around it, but wanted to check whether there is anything in the library for it first.

Thanks again for the library - take care! 👍
